### ISSUE TITLE
TeamInfoRequest is missing the optional team argument

### ISSUE THREAD

**Issue by lfbrandao** [slackapi/java-slack-sdk#715]
Hi @seratch,

I believe the TeamInfoRequest class is missing the optional `team` argument [documented here](https://api.slack.com/methods/team.info#arg_team). Because off that, I'm unable to call `methodsAsync().teamInfo` for teams other than my current team. Let me know if I'm not using this method properly. Thanks!

#### The Slack SDK version

```
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
+--- com.slack.api:bolt:1.7.0 (n)
+--- com.slack.api:slack-api-client:1.7.0 (n)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (n)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0 (n)
+--- com.slack.api:bolt-servlet:1.7.0 (n)
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
+--- com.slack.api:bolt:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0
|    +--- com.slack.api:slack-api-client:1.7.0
|    |    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    \--- com.slack.api:slack-app-backend:1.7.0
|         +--- com.slack.api:slack-api-model:1.7.0 (*)
|         +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:slack-api-model-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
+--- com.slack.api:slack-api-client-kotlin-extension:1.7.0
|    +--- com.slack.api:slack-api-model-kotlin-extension:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
+--- com.slack.api:bolt-servlet:1.7.0
|    +--- com.slack.api:slack-api-model:1.7.0 (*)
|    +--- com.slack.api:slack-api-client:1.7.0 (*)
|    +--- com.slack.api:slack-app-backend:1.7.0 (*)
|    \--- com.slack.api:bolt:1.7.0 (*)
```

#### Java Runtime version

```
openjdk version "15.0.1" 2020-10-20
OpenJDK Runtime Environment (build 15.0.1+9-18)
OpenJDK 64-Bit Server VM (build 15.0.1+9-18, mixed mode, sharing)
```

#### OS info

```
ProductName:	macOS
ProductVersion:	11.2.3
BuildVersion:	20D91
Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64
```

#### Steps to reproduce:

```
import com.slack.api.methods.MethodsClient;
import com.slack.api.methods.request.chat.ChatPostMessageRequest;
import com.slack.api.methods.response.chat.ChatPostMessageResponse;

val slack = Slack.getInstance()
val token = 'SLACK-TOKEN'
slack.methods().teamInfo { it.token(token).team('T00000000') }

```
#### Expected result:

team.info is called and returns the details of the team T00000000

#### Actual result:

teamInfo doesn't have a team property 

### Requirements

Please make sure if this topic is specific to this SDK. For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

**Comment by seratch** [comment]
@lfbrandao Thanks a lot for taking the time to report this! Indeed, the parameter is not yet supported in this SDK (the one did not exist before). If you are interested in making a contribution to this SDK, I'm happy to be a reviewer of your pull request. If you are busy and this is not urgent for you, I will work on the updates sometime soon.

**Comment by lfbrandao** [comment]
hi @seratch, hope you're well. do you have an estimate for when this will be released? it's blocking me from wrapping up a project that uses this method and I was wondering if I should work around it, or wait for a release. thanks!

**Comment by seratch** [comment]
@lfbrandao We are going to release a new patch version by the end of this week!

**Comment by seratch** [comment]
@lfbrandao I just released version 1.7.1 - please try it out 👋
